lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -443,7 +443,7 @@ def add_token_for_ff_credit(xml, payment_method, options)
       end
 
       def add_additional_3ds_data(xml, options)
-        additional_data = { 'dfReferenceId' => options[:session_id] }
+        additional_data = { 'dfReferenceId' => options[:df_reference_id] }
```

Comment:
Should we fall back to `session_id` if `df_reference_id` is not sent in the options?

Reply:
The problem is the session_id option will be a different value then we are sending today once these PRs are merged. So we can certainly do it, but it won't be the id we are currently passing here.